site/src/pages/tutorials/tutorial-blog-rest-step2.mdx

```diff
@@ -0,0 +1,109 @@
+
+Before we get into writing a blog service, first prepare a data object to contain blog post information.
+This object is to be used not only for storing blog posts, but in handling requests and sending responses.
+
+If you want a quick start, skip this step and copy off the [BlogPost.java](https://github.com/line/armeria-examples/blob/master/rest-api-annotated-service/src/main/java/example/armeria/server/blog/BlogPost.java) of the sample service code.
+
+<TutorialSteps current={2} />
+
+## What you need
+
+No preparation is required for this step.
+
+## 1. Define blog post data
+
+Our service needs a container for blog post information.
+Define a `BlogPost` class containing the following information.
+
+```java
+final class BlogPost {
+    private final int id;           // The post ID
+    private final String title;     // The post title
+    private final String content;   // The post content
+    private final long createdAt;   // The time post is created at
+    private final long modifiedAt;  // The time post is modified at
+}
+```
+
+## 2. Add constructors
+
+The sample service has the following two cases to create instances of blog posts:
+
+- [When a user creates a blog post](#constructor-for-creating-blog-posts)
+- [When a user updates a blog post](#constructor-for-updating-blog-posts)
+
+### Constructor for creating blog posts
+
+In the `BlogPost` class, add the following two constructors.
+
+```java filename=BlogPost.java
+  BlogPost(int id, String title, String content) {
+      this(id, title, content, System.currentTimeMillis());
+  }
+
+  BlogPost(int id, String title, String content, long createdAt) {
+    this(id, title, content, createdAt, createdAt);
+  }
+```
+
+### Constructor for updating blog posts
+
+Add a constructor for updating a blog post, in the `BlogPost` class.
+
+```java filename=BlogPost.java
+BlogPost(int id, String title, String content, long createdAt, long modifiedAt) {
+    this.id = id;
+    this.title = title;
+    this.content = content;
+    this.createdAt = createdAt;
+    this.modifiedAt = modifiedAt;
+}
+```
+
+## 3. Add methods
+
+Add the following block of code to use in the blog service.
+
+```java
+int id() { return id; }
+
+String title() { return title; }
+
+String content() { return content; }
+
+long createdAt() { return createdAt; }
+
+long modifiedAt() { return modifiedAt; }
+
+@Override
+public String toString() {
+    return MoreObjects.toStringHelper(this)
+                      .add("id", id)
+                      .add("title", title)
+                      .add("content", content)
+                      .add("createdAt", createdAt)
+                      .add("modifiedAt", modifiedAt)
+                      .toString();
+}
+```
+
+
+## What's next?
+
+In this step, we have implemented a data object specifying and containing blog post objects.
+
+Next, at [Step 3. Implement CREATE](./tutorial-blog-rest-step3), you'll implement a CREATE operation to create blog posts.
```

Comment:
Could we globally use paths that start with `/tutorials/tutorial-blog-rest-step3` instead of `../../path-to-page` style?

Reply:
Done! b23f747c9e5c8d2e3a2f01116ce3fe9a48f2a701
(Although, I haven't checked if all links work fine. Will make a final check on it)